csrc/multidevice/device_mesh.h

```diff
@@ -67,20 +85,42 @@ class DeviceMesh final {
     return -1;
   }
 
-  // Returns the device at a particular index in the mesh
+  // Returns the indices of a multi-dimensional mesh, or an empty vector 
+  // if device is not present
+  std::vector<int64_t> getLocalIndices(const DeviceIdxType device) const;
+
+  // Returns the device at a particular global index in the mesh
   DeviceIdxType at(int64_t index) const {
     return vector_.at(index);
   }
 
   bool operator==(const DeviceMesh& other) const {
-    return vector_ == other.vector();
+    return vector_ == other.vector() && shape_ == other.shape();
   }
 
+  // Returns the max device index in the DeviceMesh.
+  DeviceIdxType maxDeviceIdx() const;
+
+  // Returns devices of this device's team given which axis of the DeviceMesh
+  // Ex: [[0 1 2]
+  //      [3 4 5]]
+  // getTeam(4, 0) = {3, 4, 5}
+  // getTeam(4, 1) = {1, 4}
+  // TODO: calculate this once and save it
+  std::vector<DeviceIdxType> getTeam(DeviceIdxType device, int64_t axis = 0)
```

Comment:
Right now this is parameterized by axis, but we might want to change this to ParallelType, where ParallelType has to be a DID type.

Reply:
> How about updating the comment to make it more clear this is retrieving the communicator for a device given the axis?

> I'd prefer `getTeam` over `get1DSlice` since the later is describing what the output is rather than how it is implemented.

I would argue that `get1DSlice` refers to what the function *does*, not the implementation. The fact that we later use this 1D slice as a `Team` for a communication is an incidental choice that we can make, but that's not explicit nor straightforward, and actually that's not even always true:
1) in the current code, if the root is not in the mesh we append the root to the 1D slice to create the Team
2) When we will implement 2D, if, for instance, we want to bcast from mesh {0} to mesh {{0,1},{2,3}} we might wanna set the Team to be {0,1,2,3} (and root=0), so it's not a 1D slice

Does it make sense?

> How about updating the comment to make it more clear this is retrieving the communicator for a device given the axis?

I would use the word `Team` rather than `communicator`. I think suggestion has the same problem than before: it implicitely assumes that a Team is always a device mesh's 1D slice. This assumption is not necessary -- if we still decide to make such an assumption, we should state it explicitely in the comment